zeroconf/__init__.py

```diff
@@ -2978,8 +2994,9 @@ def close(self) -> None:
             if not self.unicast:
                 self.engine.del_reader(cast(socket.socket, self._listen_socket))
                 cast(socket.socket, self._listen_socket).close()
-            for s in self._respond_sockets:
-                self.engine.del_reader(s)
+            if self.unicast or self.interfaces is not InterfaceChoice.Default:
```

Comment:
Can we extract the `self.unicast or self.interfaces is not InterfaceChoice.Default` into a function or something? I think `_read_from_respond_sockets` could be a good name for it.

Reply:
I can make it a property that gets setup at `__init__`

Reply:
Yup, works for me.

Reply:
Went with `self.multi_socket` since all the conditionals would be `not self.single_socket`